numba/core/ir_utils.py

```diff
@@ -1521,15 +1521,23 @@ def find_callname(func_ir, expr, typemap=None, definition_finder=get_definition)
                 def_val = def_val._defn
             if hasattr(def_val, '__module__'):
                 mod_name = def_val.__module__
+                # The reason for first checking if the function is in NumPy's
+                # top level name space by module is that some functions are
+                # deprecated in NumPy but the functions' names are aliased with
+                # other common names. This prevents deprecation warnings on
+                # e.g. getattr(numpy, 'bool') were a bool the the target.
+                # For context see #6175, impacts NumPy>=1.20.
+                mod_not_none = mod_name is not None
+                numpy_toplevel = mod_not_none and mod_name.startswith('numpy')
```

Comment:
```suggestion
                numpy_toplevel = mod_not_none and (mod_name == 'numpy' or mod_name.startswith('numpy.'))
```
so as not to cause a warnings for `numpy_custom_module.bool`.

Reply:
Good point, done in c3bd15bc32cc2571267454b9d8b422663909de5d